docs/porting/libraries.md

```diff
@@ -1,5 +1,153 @@
+   
+This analysis phase could take some time.  You may find that you'll have to refactor your code or make significant changes to how you perform certain tasks.  You may even come to the conlcusion that porting to .NET Core isn't feasible at this time.
+
+## Choosing the project system and getting a basis for compilation
+
+Once you have analyzed the ApiPort report and your codebase, the next step is to actually begin porting your code!
+
+### Making a choice: xproj or Portable Class Library?
+
+GAWD IT FEELS SO PAINFUL WRITING THIS
+
+The first thing you'll need to do is decide on the project system you want to use.  You have two choices: `xproj` or `csproj` via a Portable Class Library (PCL).  Deciding which to pick comes down to one simple question:
+
+Do you have any need to support anything more than just having Visual Studio recognize your project?  Examples of this may be localized resources in a `.resx` folder, working seamlessly with a custom build process, and others.
+
+If yes, you should create a new PCL and select .NET Core as the target.  If you have no needs other than building and running your code in Visual Studio, you can select an xproj project type.
+
+### Getting a basis for compilation
+
+Once you've picked your project type and created a new Class Library, you'll notice that there's a new file: `project.json`.  This is the file that handles many things about your project - stating dependencies, configurations, supported target frameworks, etc. - and is used to generate a `project.lock.json` file that is used by the compiler and build system to build your project.  You can read more about `project.json` TODO: LINK_HERE.
+
+You can see what it looks like by opening it up:
+
+```javascript
+{
+    "version":"1.0.0-*",
+    
+    "dependencies":{
+        "NETStandard.Library":"1.0.0-rc2-23728"
+    },
+    
+    "frameworks":{
+        "netstandard1.4"
+    }
+}
+```
+
+The most important thing to note is the version number for `NETStandard`.  This actually corresponds to a version number for the [.NET Platform Standard](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/net-platform-standard.md), which is a standard by which you can make a clean tradeoff between API availability and target availability.  For example, if you wanted your .NET Core library to support Windows Phone targets, you would have to use `netstandard1.2`, which would cut off access to APIs available only in higher versions.
+
+You should evaluate the API availability vs Target availability tradeoff you'd like to make and pick the version of `netstandard` that fits your needs the best.  You may find that the default version picked by the template is best for your needs!
+
+Here's the essentials you need to know about `project.json`:
+
+1. You need to restore packages to generate a `project.lock.json` the first time you create the project.
```

Comment:
We should explain what that lock file is doing: I'm sure it's mysterious to most users.